### PR TITLE
Migrate Malta to `holiday_groups`, add l10n support

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Migrate Malta to `holiday_groups`, add l10n support.

Also introduce appropriate historical holiday introduction date as seen in `NATIONAL HOLIDAYS AND OTHER PUBLIC HOLIDAYS ACT` [from its 1980 amendment onwards (the earliest version online in-full)](https://legislation.mt/eli/cap/252/20210212/eng).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: tests/countries/test_malta.py [hunk 1]
```diff
@@ -9,31 +9,121 @@
...
-from datetime import date
+from holidays.countries.malta import Malta, MT, MLT
+from tests.common import TestCase
 
-import holidays
 
+class TestMalta(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(Malta, years=range(1980, 2077))
```
Comment (KJhellico) [inline tests/countries/test_malta.py:19]: If there are no holiday checks for the entire period, no need to initialize it.

**Review by KJhellico — COMMENTED** [review]
LGTM! The only thing is, maybe we should make Maltese the default language?

**Comment by KJhellico** [comment]
Later, we need to add observed holidays, absent in 2005-2020, according to [National Holidays and Other Public Holidays Act (Cap 252)](https://legislation.mt/eli/cap/252/eng).

File: README.rst [hunk 2]
```diff
@@ -440,7 +440,7 @@ The list of supported countries, their subdivisions and supported languages
    * - Malta
      - MT
      -
-     -
+     - **en_US**, mt
```
Comment (arkid15r) [inline README.rst:443]:
```suggestion
     - en_US, **mt**
```

File: holidays/countries/malta.py [hunk 3]
```diff
@@ -9,43 +9,150 @@
...
+    [Att 2 tal-2005]
+        If fall on weekends then not observed in terms of vacation leave.
+        https://legislation.mt/eli/act/2005/2/eng
+    [Att 4 tal-2021]
+        Revert Act II of 2005 changes for vacation leave.
+        https://legislation.mt/eli/cap/252/20210212/mlt
     """
 
     country = "MT"
+    default_language = "en_US"
```
Comment (arkid15r) [inline holidays/countries/malta.py:41]:
```suggestion
    default_language = "mt"
```

**Review by arkid15r — COMMENTED** [review]
+1 here, it should be at least en_MT

Other than that looks great!

File: tests/countries/test_malta.py (hunk 1 above)
Comment (PPsyrius) [reply tests/countries/test_malta.py:19]: Good to know 👀

**Comment by PPsyrius** [comment]
> +1 here, it should be at least en_MT

Agree - I'm making the changes ATM, IMO for standardization purposes Canada's `en` should also probably be changed to `en_US` IMO

**Comment by PPsyrius** [comment]
> Later, we need to add observed holidays, absent in 2005-2020, according to [National Holidays and Other Public Holidays Act (Cap 252)](https://legislation.mt/eli/cap/252/eng).

IMO unless I read wrong, the 2005 and 2021 amendments to the law have no effect on the current implementation since it's still a holiday, no in lieus are added or removed, just that it no longer counts as paid holidays in terms of wages or vacation leaves between 2005-2020 as seen [here](https://legislation.mt/eli/act/2005/2/eng)
image

**Comment by KJhellico** [comment]
> IMO unless I read wrong, the 2005 and 2021 amendments to the law have no effect on the current implementation since it's still a holiday, no in lieus are added or removed

Yes, you're right! 👍 Day off is not transferred to Monday, but simply added to the vacation.

**Review by KJhellico — COMMENTED** [review]
LGTM!

File: tests/countries/test_malta.py [hunk 4]
```diff
@@ -9,31 +9,121 @@
...
+            ("2023-06-29", "Il-Festa ta' San Pietru u San Pawl"),
+            ("2023-08-15", "Il-Festa ta' Santa Marija"),
+            ("2023-09-08", "Jum il-Vitorja"),
+            ("2023-09-21", "Jum l-Indipendenza"),
+            ("2023-12-08", "Il-Festa tal-Immakulata Kunċizzjoni"),
+            ("2023-12-13", "Jum ir-Repubblika"),
+            ("2023-12-25", "Il-Milied"),
+        )
+
+    def test_l10n_en_MT(self):
```
Comment (arkid15r) [inline tests/countries/test_malta.py:112]:
```suggestion
    def test_l10n_en_mt(self):
```

**Review by arkid15r — COMMENTED** [review]
🇲🇹 LGTM

**Comment by arkid15r** [comment]
Merged into beta!

Thank you!

**Pull request merged**